core/service/src/lib.rs

```diff
@@ -152,7 +152,7 @@ impl<Components: components::Components> Service<Components> {
 
 		let (client, on_demand) = Components::build_client(&config, executor)?;
 		let select_chain = Components::build_select_chain(&mut config, client.clone())?;
-		let import_queue = Box::new(Components::build_import_queue(
+		let import_queue = Arc::new(Components::build_import_queue(
```

Comment:
I don't think we ever clone this `Arc`. IMO it would be preferable to keep the `Box` but remove `ImportQueueClone`. This way we guarantee that nothing else but the network code can access the import queue, and we can start introducing `&mut` methods.